File: Master_the_Key_ROS_Concepts/Lec1_Create_a_Cakin_Workspace/catkin_ws/src/my_robot_tutorials/scripts/add_two_ints_client.py
#!/usr/bin/env python3

import rospy
from rospy_tutorials.srv import AddTwoInts


if __name__ == '__main__':
    rospy.init_node("add_two_ints_client")

    # Need to create service client
    rospy.wait_for_service("/add_two_ints") # /name of the service   where it advertise

    # Create the client to call the service
    # If the client fail the client server, it will need exception
    try:
        # Call the service
        # Create client
        add_two_ints = rospy.ServiceProxy("/add_two_ints", AddTwoInts)
        response = add_two_ints(2,6) #call the client (2,6) is where a and b
        rospy.loginfo("Sum is : " +str(response.sum))
    except rospy.ServiceExcetion as e:
        rospy.lowarn("Service Failed: "+ str(e))

File: Master_the_Key_ROS_Concepts/Lec1_Create_a_Cakin_Workspace/catkin_ws/src/my_robot_tutorials/src/add_two_ints_client.cpp
#include <ros/ros.h>
#include <rospy_tutorials/AddTwoInts.h>


int main(int argc, char **argv)
{
	ros::init(argc, argv, "add_two_ints_client");
	ros::NodeHandle nh;

	// Create service client
	ros::ServiceClient client = nh.serviceClient<rospy_tutorials::AddTwoInts>("/add_two_ints");

	// Directly create message
	rospy_tutorials::AddTwoInts srv;
	srv.request.a = 12;
	srv.request.b = 5;

	// call function client.call(service)
	if (client.call(srv)){
		//process data if True
		ROS_INFO("Returned sum is %d", (int)srv.response.sum);
	}
	else {
		ROS_WARN("SERVICE CALL FAILED");	
	}
}
